### ISSUE TITLE
[SDK 49] react-native-maps crashes app with location permission

### ISSUE THREAD

**Issue by ChromeQ** [expo/expo#23396]
### Minimal reproducible example

https://github.com/ChromeQ/expo-sdk49-maps

### Summary

I have an existing app working well with maps on expo-sdk48. After upgrade to sdk49 the app crashes on load.

I have found that the app only crashes when location permission is turned on, and when react-native-maps uses the `showUserLocation` prop, and when in development build (and probably production build). Interestingly Expo Go does work.
I have not tested iOS.

I have created a minimal reproduction repo - https://github.com/ChromeQ/expo-sdk49-maps
The main branch is the initial commit from a simple `create-expo-app`
I created several branches with PR's to easily review the difference and progressively upgraded react-native-maps to latest, then expo sdk to 49 where it crashes on load.

I believe this is a bug within expo rather than react-native-maps, as this new sdk is new (react-native-maps@1.7.1 has been stable since 23rd April. And expo sdk 48 also works with the latest maps version. The only change between my branches is expo upgrade (and the required dependencies it requires)

### Environment

❯ npx expo-env-info

  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 18.16.1 - ~/.nvm/versions/node/v18.16.1/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 9.5.1 - ~/.nvm/versions/node/v18.16.1/bin/npm
    SDKs:
      Android SDK:
        Android NDK: 25.2.9519653
    npmPackages:
      expo: ^49.0.0 => 49.0.0 
      react: 18.2.0 => 18.2.0 
      react-native: 0.72.1 => 0.72.1 
    Expo Workflow: managed

**Comment by ChromeQ** [comment]
The app crashes with `java.lang.RuntimeException: Unable to resume activity` and `Caused by java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.FusedLocationProviderClient, but class was expected` seems to be the main culprit.

<details>
  <summary>The full error stack trace</summary>
  
 ```
 ERROR  Your app just crashed. See the error below.
java.lang.RuntimeException: Unable to resume activity {com.chromeq.exposdk49maps/com.chromeq.exposdk49maps.MainActivity}: java.lang.reflect.InvocationTargetException
  android.app.ActivityThread.performResumeActivity(ActivityThread.java:4773)
  android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4806)
  android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
  android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
  android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
  android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
  android.app.ActivityThread$H.handleMessage(ActivityThread.java:2306)
  android.os.Handler.dispatchMessage(Handler.java:106)
  android.os.Looper.loopOnce(Looper.java:201)
  android.os.Looper.loop(Looper.java:288)
  android.app.ActivityThread.main(ActivityThread.java:7918)
  java.lang.reflect.Method.invoke(Native Method)
  com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
  com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by java.lang.reflect.InvocationTargetException
  java.lang.reflect.Method.invoke(Native Method)
  expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod(ReactActivityDelegateWrapper.kt:274)
  expo.modules.ReactActivityDelegateWrapper.onResume(ReactActivityDelegateWrapper.kt:152)
  com.facebook.react.ReactActivity.onResume(ReactActivity.java:58)
  android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1564)
  android.app.Activity.performResume(Activity.java:8474)
  android.app.ActivityThread.performResumeActivity(ActivityThread.java:4763)
  android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4806)
  android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
  android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
  android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
  android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
  android.app.ActivityThread$H.handleMessage(ActivityThread.java:2306)
  android.os.Handler.dispatchMessage(Handler.java:106)
  android.os.Looper.loopOnce(Looper.java:201)
  android.os.Looper.loop(Looper.java:288)
  android.app.ActivityThread.main(ActivityThread.java:7918)
  java.lang.reflect.Method.invoke(Native Method)
  com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
  com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.FusedLocationProviderClient, but class was expected (declaration of 'com.google.android.gms.location.FusedLocationProviderClient' appears in /data/app/~~VOCIH5e3WgYMx6lMYYEu5g==/com.chromeq.exposdk49maps-WO1q4VNP9CUd5w5AJxosVw==/base.apk)
  com.rnmaps.maps.FusedLocationSource.activate(FusedLocationSource.java:49)
  com.google.android.gms.maps.zzs.activate(com.google.android.gms:play-services-maps@@18.0.0:1)
  com.google.android.gms.maps.internal.zzj.zza(com.google.android.gms:play-services-maps@@18.0.0:6)
  com.google.android.gms.internal.maps.zzb.onTransact(com.google.android.gms:play-services-maps@@18.0.0:3)
  android.os.Binder.transact(Binder.java:1164)
  m.fe.c(:com.google.android.gms.dynamite_mapsdynamite@232414044@23.24.14 (190400-0):2)
  com.google.android.gms.maps.internal.l.e(:com.google.android.gms.dynamite_mapsdynamite@232414044@23.24.14 (190400-0):0)
  com.google.maps.api.android.lib6.impl.bk.F(:com.google.android.gms.dynamite_mapsdynamite@232414044@23.24.14 (190400-0):6)
  com.google.android.gms.maps.internal.i.bb(:com.google.android.gms.dynamite_mapsdynamite@232414044@23.24.14 (190400-0):204)
  m.ff.onTransact(:com.google.android.gms.dynamite_mapsdynamite@232414044@23.24.14 (190400-0):4)
  android.os.Binder.transact(Binder.java:1164)
  com.google.android.gms.internal.maps.zza.zzc(com.google.android.gms:play-services-maps@@18.0.0:2)
  com.google.android.gms.maps.internal.zzg.setLocationSource(com.google.android.gms:play-services-maps@@18.0.0:3)
  com.google.android.gms.maps.GoogleMap.setLocationSource(com.google.android.gms:play-services-maps@@18.0.0:2)
  com.rnmaps.maps.MapView$15.onHostResume(MapView.java:424)
  com.facebook.react.bridge.ReactContext.onHostResume(ReactContext.java:308)
  com.facebook.react.ReactInstanceManager.moveToResumedLifecycleState(ReactInstanceManager.java:778)
  com.facebook.react.ReactInstanceManager.onHostResume(ReactInstanceManager.java:677)
  com.facebook.react.ReactInstanceManager.onHostResume(ReactInstanceManager.java:627)
  com.facebook.react.ReactDelegate.onHostResume(ReactDelegate.java:70)
  com.facebook.react.ReactActivityDelegate.onResume(ReactActivityDelegate.java:123)
  java.lang.reflect.Method.invoke(Native Method)
  expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod(ReactActivityDelegateWrapper.kt:274)
  expo.modules.ReactActivityDelegateWrapper.onResume(ReactActivityDelegateWrapper.kt:152)
  com.facebook.react.ReactActivity.onResume(ReactActivity.java:58)
  android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1564)
  android.app.Activity.performResume(Activity.java:8474)
  android.app.ActivityThread.performResumeActivity(ActivityThread.java:4763)
  android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4806)
  android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
  android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
  android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
  android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
  android.app.ActivityThread$H.handleMessage(ActivityThread.java:2306)
  android.os.Handler.dispatchMessage(Handler.java:106)
  android.os.Looper.loopOnce(Looper.java:201)
  android.os.Looper.loop(Looper.java:288)
  android.app.ActivityThread.main(ActivityThread.java:7918)
  java.lang.reflect.Method.invoke(Native Method)
  com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
  com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
  ```
</details>

**Comment by alenquer** [comment]
I fixed this error by changing the version of react-native-maps to 2.0.0-beta.14.

The issue has been fixed in version [v.2.0.0-beta.10](https://github.com/react-native-maps/react-native-maps/releases/tag/v2.0.0-beta.10
)

**Comment by ChromeQ** [comment]
> I fixed this error by changing the version of react-native-maps to 2.0.0-beta.14.
> 
> The issue has been fixed in version [v.2.0.0-beta.10](https://github.com/react-native-maps/react-native-maps/releases/tag/v2.0.0-beta.10)

I can confirm my app is working as expected with the beta version of react-native-maps with no regressions so far. 
Also new with expo sdk 49 is the ability to ignore dependency checking for `expo-doctor` or `expo install --check`, just add this to your package.json

```json
"expo": {
    "install": {
      "exclude": [
        "react-native-maps"
      ]
    }
  }
```

More info: https://docs.expo.dev/more/expo-cli/#configuring-dependency-validation

**Comment by brentvatne** [comment]
thanks for following up! i'm hesitant to change the recommended version of react-native-maps to a beta release. ideally we can get a patch release of react-native-maps@1.7 that fixes this

**Comment by ChromeQ** [comment]
I agree, I would also raise an eyebrow if I saw a dependency on a beta.

Yes ideally it would be fixed on react-native-maps, but they have been stable for a while and I assume a bump of the `com.google.android.gms:play-services-location` dependency is potentially a breaking change to some projects (as we are finding out) which is why I assume react-native-maps went to a v2.0.0 beta rather than break projects who use `~` or `^` package.json dependency.

As I said on the [other thread](https://github.com/expo/expo/pull/22468#issuecomment-1627377933):
> the whole point of Expo is to guarantee (as much as possible) that a known version of set of dependencies work as expected with the SDK. And the SDK 49 upgrades react-native-maps to 1.7.1 which has this known limitation.

It does feel like expo favours iOS and didn't test a whole lot on android as this is not the first "big bug" I've come across in the last few days - and they all affect android only

**Comment by forki** [comment]
We're hit by this as well. This is really unfortunate and should be mentioned in the sdk 49 blog post.

**Comment by forki** [comment]
Is there a way to fix this by expo downgrading the maps dependency for now?

**Comment by forki** [comment]
![Screenshot_2023-07-11-22-03-57-686_com google android gm](https://github.com/expo/expo/assets/57396/d2e85920-5fdb-4d0d-a7e4-b3abbe3c2375)
We tried to update to 2.0 beta as described. But Google  isn't happy with that.

**Comment by brentvatne** [comment]
@ChromeQ - i appreciate all of your research into this! thank you.

> the whole point of Expo is to guarantee (as much as possible) that a known version of set of dependencies work as expected with the SDK. And the SDK 49 upgrades react-native-maps to 1.7.1 which has this known limitation.

this is indeed one of the benefits that we aim to offer. as you can imagine, with the surface area of the expo sdk, it's not always possible to cover every case. in https://github.com/expo/expo/pull/22908, @aleqsio followed our qa process to test react-native-maps within this repo, but this doesn't always work exactly the same as other projects and we don't exercise the entirety of the API for every third party module. for third party modules, it's more of an smoke test, and we trust that those modules follow their own processes for qa before releasing. our smoke test did not include the `showsUserLocation` prop. we had a beta period for a week and this was also not encountered at that time. immediately after a release, we often uncover new issues that were not surfaced in qa or beta.

in the future, if you wait a couple weeks after a release, these types of issues will likely be handled already, or warnings will be in place to let folks know about them. we missed this particular regression and i'm sure we'll miss some in the future as well. our response in these types of situations is to try to handle the issue quickly, while also empowering developers to work around the issue on their own, rather than being blocked on us. i'm happy that you've identified a workaround and have this resolved on your end. @gabrieldonadel is currently looking into the best way for us to solve this on our end, and i've added a note to the "Known Regressions" in the release notes.

thanks again for the report and all of the followup here.

**Comment by brentvatne** [comment]
> We tried to update to 2.0 beta as described. But Google isn't happy with that.

@forki - can you elaborate on what you mean by this?

**Comment by forki** [comment]
@brentvatne I mean that as you can see in the screenshot the workaround seems to fail on some devices. Before the upgrade to 2.0 beta we did not have any errors on the Google play store prerelease test. Now we do. So in other words: the workaround is not working properly. I assume the proper fix for now is a downgrade

**Comment by ChromeQ** [comment]
@brentvatne Thanks for the insight to the testing process, and yes I know it must be difficult to test everything and especially when it comes to 3rd parties, and I certainly appreciate all the work you guys do as I certainly wouldn't be close to building an app and deploying it without Expo.

I don't want to complain, especially as I like to be always up to date and bleeding edge so I am used to finding workarounds or breaking stuff, but this was particularly frustrating as it seemed like such a fundamental and easy to reproduce bug resulting in a proper crash and the problem looked to be out of my area of expertise (I'm a ts/js developer), but I am also happy to contribute to file issues and find and test out solutions. Perhaps the smoke tests should now include this 😛

Looking forward to seeing a proper fix soon as. Thanks

**Comment by ChromeQ** [comment]
@forki I checked out my google prerelease report and there are no issues with any device. Not sure how simple of a project you have or your specific usage of maps, but it is possible that other issues are arising from other props for example (much like this one is only arising when using `showUserLocation`, and make sure you are using at least v2.0.0-beta.10 or 14 is latest.

The fact that this bug caused a total crash of the app as soon as the map loads and _some_ of your devices passed suggest this bug is not the only one affecting your project

**Comment by forki** [comment]
I downgraded the dependency to the one that was used in sdk 48. IIRC that was 1.3.2 and all reported issues from the Google play store prerelease report are gone. So I think I will go with that for now.

**Comment by lucaskab** [comment]
In my case, using any Beta version or downgrading to 1.3.2 the app still crashes. I couldn't find any solution to make it work at least once after upgrading to SDK 49

**Comment by gabrieldonadel** [comment]
A new version of `expo-location` fixing this issue has just been published to npm 

Please upgrade to v16.1.0 by running `yarn add expo-location@16.1.0` or `npx expo install --fix` 

Happy Coding! 🎉

**Comment by ljmzbigsmart** [comment]
Im having this exact same issue, but only in bundle release app that is downloaded from Play Store. Running the app with the bundler or with debug APK doesn´t do this.
react-native-maps 1.7.1
expo-location 16.1.0
sdk 49

**Comment by Swapnil-trinesis** [comment]
@ljmzbigsmart Even I am facing similar Issue. I am using same versions you mentioned.

**Comment by niug** [comment]
I can't correct this issue with the beta version of react-native-maps. If I use the beta when I run 'expo start' the app crashes because can't find RNMMap component. My package.json is:

```
{
  "main": "expo-router/entry",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@react-native-async-storage/async-storage": "1.18.2",
    "expo": "~49.0.5",
    "expo-constants": "~14.4.2",
    "expo-linking": "~5.0.2",
    "expo-router": "2.0.0",
    "expo-status-bar": "~1.6.0",
    "firebase": "^10.2.0",
    "latlon-geohash": "^2.0.0",
    "react": "18.2.0",
    "react-native": "0.72.6",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-maps": "2.0.0-beta.14",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-svg": "13.9.0",
    "react-native-web": "~0.19.6",
    "expo-image-picker": "~14.3.2",
    "expo-location": "16.1.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0"
  },
  "private": true,
  "expo": {
    "install": {
      "exclude": [
        "react-native-maps"
      ]
    }
  }
}
```

And crashes shows following error:  
```ERROR  Invariant Violation: requireNativeComponent: "RNMMap" was not found in the UIManager.```

Someone can help me?
Thanks!

**Comment by ChromeQ** [comment]
@niug The fix I outlined above with upgrading `react-native-maps` to beta version will only work with `expo-location@16.0.0` -- expo location updated to avoid using the beta of maps. 

You can downgrade react-native-maps to 1.7.1 and keep expo-location on 16.1.0
or
You can keep react-native-maps on the beta and downgrade expo-location to 16.0.0

**Comment by Swapnil-trinesis** [comment]
@niug 
I had faced similar Issue.
Updating my android/build.gradle with below changes solved it for me.
you can try this solution.

`allprojects {
....
    subprojects {
        project.configurations.all {
            resolutionStrategy.eachDependency { details ->
                if (details.requested.group == 'com.google.android.gms'
                        && details.requested.name == 'play-services-location') {
                    details.useVersion '20.0.0'
                }
            }
        }
    }
}`

**Comment by niug** [comment]
Thank you very much for the quick responses.

@ChromeQ I have tried the two solutions you propose, and when I run 'npm run android' the following happens with each proposal:
 - With react-native-maps 1.7.1 and expo-location 16.1.0 expo restarts without giving any message.
 - With react-native-maps beta and expo-location 16.0.0.0 the RNMMap error occurs.

I don't know if I have to configure something else in the project.

@Swapnil-trinesis this problem occurs when I run with 'npm run android', which launches the 'expo start', I do not know where to put the file that you tell me. I have created it in the root but it doesn't seem to get it.

**Comment by ChromeQ** [comment]
@niug I am in a managed expo app, so I don't know how to fix it from inside the android folder unfortunately

**Comment by Swapnil-trinesis** [comment]
@niug 
You don't have to create a new build.gradle file in root. It's already present in your React native project's android folder. You just need to modify it.

**Comment by karimdakroury** [comment]
> @niug The fix I outlined above with upgrading `react-native-maps` to beta version will only work with `expo-location@16.0.0` -- expo location updated to avoid using the beta of maps.
> 
> You can downgrade react-native-maps to 1.7.1 and keep expo-location on 16.1.0 or You can keep react-native-maps on the beta and downgrade expo-location to 16.0.0

@ChromeQ Hi, after updating react-native-maps from 1.8.0 to 1.7.1 and expo-location from 14.3.0 to 16.1.0. i started getting this error Error: Cannot find module 'expo/config-plugins' after checking the package the log was referencing i noticed that in my expo-location v14.3.0 it was using const config_plugins_1 = require("@expo/config-plugins"); and in v16.1.0 it is using const config_plugins_1 = require("expo/config-plugins"); hence the error. What should i do?

**Comment by niug** [comment]
@Swapnil-trinesis I can't find the android folder inside the project, I think it's because I use the expo library, which manages the android and iOS folders. If I don't have this folder, there is some option to manage this build.gradle

**Comment by ChromeQ** [comment]
@niug if you don't have an android or iOS folder then you have what is called an expo managed project. Nothing wrong with that, this bug has been fixed with the latest supported versions of expo-location and react-native-maps.
Which version of these do you have installed?

**Comment by niug** [comment]
@ChromeQ , Sorry, I finally found the bug. My problem was that I was trying to get the current location with expo-location in the onLoad component (with useEffect) at the same time as MapView was loading the user's location. When I called getLocation() in useEffect the application crashed.

By removing the getLocation in useEffect the Map runs correctly and displays the user's location correctly.

Thank you very much for your quick responses.

**Comment by frdev** [comment]
Hi guys, I need help with expo-location package.

## Minimal reproducible example
When the function getCurrentPositionAsync() is called, I'm getting the error below on Android. I tried to update/downgrade the dependencies, but didn't work.

## Summary

My custom useGeolocation hook:

```ts
import * as Location from "expo-location";

export function useGeolocation() {
  const getPosition = async (
    options?: Location.LocationOptions
  ): Promise<Location.LocationObjectCoords> => {
    const { status } = await Location.requestForegroundPermissionsAsync();

    if (status !== "granted") throw new Error("Permission denied");

    const { coords } = await Location.getCurrentPositionAsync(options);

    return coords;
  };

  return {
    getPosition,
  };
}
```

Error:

java.lang.reflect.InvocationTargetException
java.lang.reflect.Method.invoke(Native Method)
expo.modules.core.ExportedModule.invokeExportedMethod(ExportedModule.java:98) expo.modules.adapters.react.NativeModulesProxy.callMethod(NativeModulesProxy.java:178)
expo.modules.kotlin.defaultmodules.NativeModulesProxyModule$definition$lambda-1$$inlined$AsyncFunction$1.invoke(ObjectDefinitionBuilder.kt:357)
expo.modules.kotlin.defaultmodules.NativeModulesProxyModule$definition$lambda-1$$inlined$AsyncFunction$1.invoke(ObjectDefinitionBuilder.kt:244)
expo.modules.kotlin.functions.AsyncFunctionWithPromiseComponent.callUserImplementation$expo_modules_core_debug(AsyncFunctionWithPromiseComponent.kt:19)
expo.modules.kotlin.functions.AsyncFunction$attachToJSObject$2$1.invokeSuspend(AsyncFunction.kt:69)
kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
android.os.Handler.handleCallback(Handler.java:938)
android.os.Handler.dispatchMessage(Handler.java:99)
android.os.Looper.loop(Looper.java:223)
android.os.HandlerThread.run(HandlerThread.java:67)
java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.FusedLocationProviderClient, but class was expected (declaration of 'com.google.android.gms.location.FusedLocationProviderClient' appears in /data/app/~~khZkjIQjaV-ejjoLYq47mQ==/br.com.manuteno-KpmBo_dV2LgrdEiu0mXqkw==/base.apk!classes9.dex)
expo.modules.location.LocationModule.getCurrentPosition(LocationModule.java:620)
expo.modules.location.LocationModule.getCurrentPositionAsync(LocationModule.java:262)

## Dependencies:

```
"react-native": "0.70.8",
"react-native-maps": "^1.7.1",
"expo": "~47.0.14",
"expo-localization": "~14.0.0",
"expo-location": "~15.0.1",
```

How I can fix that?

## Environment
```
expo-env-info 1.0.5 environment info:
System:
OS: macOS 13.4.1
Shell: 5.9 - /bin/zsh
Binaries:
Node: 18.17.0 - ~/.nvm/versions/node/v18.17.0/bin/node
Yarn: 1.22.17 - /usr/local/bin/yarn
npm: 9.6.7 - ~/.nvm/versions/node/v18.17.0/bin/npm
Managers:
CocoaPods: 1.11.2 - /usr/local/bin/pod
SDKs:
iOS SDK:
Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
IDEs:
Android Studio: 2022.3 AI-223.8836.35.2231.11005911
Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
npmPackages:
expo: ~47.0.14 => 47.0.14
react: 18.1.0 => 18.1.0
react-native: 0.70.8 => 0.70.8
Expo Workflow: bare
```

**Comment by ChromeQ** [comment]
@frdev is this a new project or an existing one where you are stuck on expo sdk 47?
SDK49 is the latest so I'd recommend updating if possible.

**Comment by kbothe** [comment]
> I fixed this error by changing the version of react-native-maps to 2.0.0-beta.14.
> 
> The issue has been fixed in version [v.2.0.0-beta.10](https://github.com/react-native-maps/react-native-maps/releases/tag/v2.0.0-beta.10)

how do you switch the package to the beta version? i'm currently having the same issue.

**Comment by jlsnjnr** [comment]
Guys, besides the SHA-1 settings, what helped me fix the behavior was changing the 'initialRegion' to just 'region'
`<MapView 
  loadingEnabled 
  provider={PROVIDER_GOOGLE} 
  ref={mapRef} 
  className="w-full h-screen flex-1" 
  region={{ 
  latitude: -23.179363, 
  longitude: -45.885344, 
  latitudeDelta: 0.01, 
  longitudeDelta: 0.01 
  }} 
/>``

**Comment by KeeJin** [comment]
@ljmzbigsmart Have you managed to resolve your issue? I seem to be encountering a similar issue: 
My app works fine in expo go, but crashes when built as an apk.

**Comment by KJCheo-weston** [comment]
> @ljmzbigsmart Have you managed to resolve your issue? I seem to be encountering a similar issue: My app works fine in expo go, but crashes when built as an apk. Expo version: v50.0.14 Expo-location version: v16.5.5 React-native-maps version: v1.10.0

Sorry I should have done a more thorough check, my problem was with me not setting up an API key for using Google's Maps SDK for Android. I wrongly assumed that I didn't need it since it works in Expo Go, when in reality Expo Go probably maintains its own API keys for such services.

**Comment by abhey-gupta** [comment]
> @ljmzbigsmart Have you managed to resolve your issue? I seem to be encountering a similar issue: My app works fine in expo go, but crashes when built as an apk. Expo version: v50.0.14 Expo-location version: v16.5.5 React-native-maps version: v1.10.0

I am also facing the same issue. It works fine on ios, but in android it crashes in production. I have correctly set up the google api keys but still the issue persists. All the packages are up to date as of now

Any help would be appreciated

expo version: 50.0.15
Expo-location version: 16.5.5
React-native-maps version: 1.10.0

**Comment by sergiokano** [comment]
Exactly, I had the same first issue @KJCheo-weston 

**All of you have to check you are using a google maps API KEY**

Expo gives you an api key, but when you build the apk are out of expo

My problem was i had not the api key, but now is not working anyway, I am gonna try to change to beta version

**Comment by 4rexdadi** [comment]
> Exactly, I had the same first issue @KJCheo-weston 
> 
> 
> 
> **All of you have to check you are using a google maps API KEY**
> 
> 
> 
> Expo gives you an api key, but when you build the apk are out of expo
> 
> 
> 
> My problem was i had not the api key, but now is not working anyway, I am gonna try to change to beta version

Where you able to fix this and what was your solution?

**Comment by inoublileith** [comment]
I am also facing the same issue.

Any help would be appreciated

expo version: 49.0.6
expo-location version: ~16.1.0
react-native-maps version: 1.7.0

**Comment by Kaizer228** [comment]
Has anyone solved the issue with react-native-maps? It works fine in my Expo Go, but it crashes when I build the APK. I suspect that the API key might be the issue. Does anyone know how to set up the API key for react-native-maps?, i stuck in this issue thaks

**Comment by cyrusmcc** [comment]
> Has anyone solved the issue with react-native-maps? It works fine in my Expo Go, but it crashes when I build the APK. I suspect that the API key might be the issue. Does anyone know how to set up the API key for react-native-maps?, i stuck in this issue thaks

Any luck resolving this? Same issue here.

**Comment by ChromeQ** [comment]
This issue was not about API key issues so why do comments keep getting added to it? The initial issue is resolved with the latest versions.

If your issue is the map crashes in a development/production build whilst it works in Expo Go, then it sounds like an API key issue. Please make sure you follow the instructions on [Expo Map View](https://docs.expo.dev/versions/latest/sdk/map-view/#add-the-api-key-to-your-project). Specifically about where to add it to your expo app.config as it is slightly different in iOS and android.

![image](https://github.com/user-attachments/assets/0acb80ed-7825-42c6-9afa-f36880163afc)

If there are other issues around the map crashing even with API keys defined correctly then open another issue, this one is closed for a reason, it is resolved.

**Comment by PaneruVipin** [comment]
The issue you're facing occurs because react-native-maps requires additional configuration when building for a production APK. It works fine in the development environment (local), but for the APK, you need to configure the Google Maps SDK for Android. Here's a detailed breakdown of the steps you need to follow:

### 1. **Register a Google Cloud API Project**
   - Go to the [Google Cloud Console](https://console.cloud.google.com/).
   - Create a new project or use an existing one.
   - Enable the **Maps SDK for Android** by navigating to **APIs & Services > Library**, searching for **Maps SDK for Android**, and enabling it.

### 2. **Create an API Key**
   - In the [Google Cloud Console](https://console.cloud.google.com/), go to **APIs & Services > Credentials**.
   - Click on **Create Credentials** and select **API Key**.
   - Initially, **do not restrict** the API key (you can restrict it later if needed).
   - Copy the API key that is generated.

### 3. **Add the API Key to Your Project**
   - In your **app.json** (or **app.config.js**) file, add the API key in the appropriate section for Android:
     ```json
     "android": {
       "config": {
         "googleMaps": {
           "apiKey": "YOUR_GOOGLE_MAPS_API_KEY"
         }
       }
     }
     ```
   - Replace `"YOUR_GOOGLE_MAPS_API_KEY"` with the actual API key you generated.

### 4. **Update `react-native-maps` to Use Google Maps**
   - In your code, ensure you import the `PROVIDER_GOOGLE` from `react-native-maps` and pass it to the `MapView` component:
     ```javascript
     import MapView, { PROVIDER_GOOGLE } from 'react-native-maps';

     <MapView
       provider={PROVIDER_GOOGLE}
       style={{ flex: 1 }}
       region={mapRegion}
     >
       <Marker coordinate={mapRegion} />
     </MapView>
     ```

### 5. **Rebuild Your APK**
   - After adding the API key and making the necessary configurations, rebuild your APK.
   - If you're using **Expo**, make sure you're using the appropriate build configuration, such as **EAS Build**.
   - For **bare React Native projects**, run the following command to build your APK:
     ```bash
     npx react-native run-android --variant=release
     ```

### 6. **Test the APK**
   - Once the APK is built and installed on a physical device, test the app to ensure the map is loading correctly.
   - If everything works fine, you can now choose to **restrict** the API key for additional security.

### 7. **Restrict the API Key (Optional but Recommended)**
   - If you want to restrict the API key later for security reasons, go back to **Google Cloud Console > APIs & Services > Credentials**.
   - Click on your API key and under **Key restrictions**, choose **Android apps**.
   - Add your **package name** (e.g., `com.yourapp.name`) and the **SHA-1 certificate** (if you plan to restrict it to a specific certificate).
     - To get the **SHA-1 certificate** for development builds, run:
       ```bash
       keytool -list -v -keystore ~/.android/debug.keystore -alias androiddebugkey -storepass android -keypass android
       ```
     - For **Google Play Store builds**, retrieve the SHA-1 from the **App Integrity** section in the **Google Play Console**.

### 8. **Rebuild and Test Again**
   - After restricting the API key, rebuild your APK and test again to ensure everything is working correctly with the restricted API key.

By following these steps, you can successfully configure Google Maps for your React Native app without restricting the API key initially, and later add the restrictions for security if needed.
